### ISSUE TITLE
Allow users to add/remove courses and categories

### ISSUE THREAD

**Issue by SergeyKhval** [open-source-society/ossu-client#29]
Basically user should be able to add/remove any course from curriculum to their profile.

Whole category?
May be also possibility to add links to any MOOC which will be added to "Additional courses" section

**Comment by crmackay** [comment]
:+1: on this...it does seem difficult to implement though...

**Comment by ericdouglas** [comment]
Practical example when to removing a whole category makes sense:

The student already has a degree in CS or related area and want to take some OSSU classes. In this case, the student should be able to remove some courses/categories that he/she already took.

:+1: to add an "Additional Courses" section.

**Summarizing** (user will be able to):

- add or remove a course in some existing category
- add or remove a course in the "Additional Courses" category
- remove a category

I don't know if it's a good idea give the possibility to add categories. Maybe add courses in the "Additional courses" section will suffice.

I would love to know your opinions about those ideas.

**Comment by rodrigobaron** [comment]
:+1:
I think when a user open the curriculum he can see:
* grade courses
* extra courses
* additional courses

...awesome!

but since this is an open course all categories and courses (grade and extra) should be managed by community. Because the public user profile will be more solid.

what do you think?

**Comment by ericdouglas** [comment]
@rodrigobaron 

I didn't understand if you are suggesting that the community should be able to manage the main curriculum (the "source of truth"), or change just their "fork" of the main curriculum.

Our curriculum shouldn't be that "manageable". We need to keep it as immutable as possible because remove a course that you already did is disappointing, as well as add more courses in one category that you already have completed.

The extras section have the function to be a more democratic place, where students and collaborators, in general, are able to add courses.

Everyone can suggest an alteration in the main curriculum, but we need to be much more careful to make changes in such file.

The feature discussed here will give students the ability to manipulate their fork of the main curriculum, but we will keep the main version "immutable" as far as we can.

In that fork the student will be able to add extras courses in the existent categories and for courses that are not related with any of the existent categories, he/she will add the course in the "Additional Courses" category.

**Comment by crmackay** [comment]
so how about this:

- users cannot delete any "main curriculum"...if you took a similar course, or otherwise think you don't need to take it, you can enter "complete" and add a link to a file on github demonstrating or explaining that completion/exemption etc...
     - example: I took Calculus 1, and Calculus 2 in highschool, so for those courses, I could link to a file on github with the details (maybe a markdown file?)

- users can add "extra" courses to any category they want
    - example: user might add ["Approximation Algorithms"](https://www.coursera.org/learn/approximation-algorithms-part-1) to the *Algorithms* section

- users can add new categories that would be like "specializations"
  - example: I would add "Bioinformatics", and list the 4 or 5 courses I want to take that are specific to Bioinformatics

**Comment by ericdouglas** [comment]
@crmackay 

> "users cannot delete any "main curriculum"...if you took a similar course, or otherwise think you don't need to take it, you can enter "complete" and add a link to a file on github demonstrating or explaining that completion/exemption etc..."

It's an interesting approach, student should at least write an article to show that he/she concluded a equivalent course. The ideal situation would be the student create a project, but an article works too, for example.

> "users can add "extra" courses to any category they want"

Perfect! :+1: 

> "users can add new categories that would be like "specializations""

Yes, this makes a lot of sense. For situations like this one I particularly think that "add categories" will be a nice feature.

**Comment by SergeyKhval** [comment]
@crmackay I was thinking in the same direction. I suppose we can use this approach as a starting point

**Comment by rodrigobaron** [comment]
@ericdouglas yes, main courses and extras are "moderate" by community, and additional by users... I was thinking not to let the user add/remove categories but it makes sense when using for specializations like @crmackay said ...
